Fix:

configure:3234:7: error: implicitly declaring library function 'exit' with type 'void (int) __attribute__((noreturn))' [-Werror,-Wimplicit-function-declaration]

index_main.c:84:7: error: implicitly declaring library function 'strcmp' with type 'int (const char *, const char *)' [-Werror,-Wimplicit-function-declaration]

index_main.c:241:6: error: implicit declaration of function 'index_query' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
--- configure.orig	2004-11-19 12:19:06.000000000 -0600
+++ configure	2021-02-25 16:35:28.000000000 -0600
@@ -3212,6 +3212,7 @@
 cat >>conftest.$ac_ext <<_ACEOF
 /* end confdefs.h.  */
 #include <ctype.h>
+#include <stdlib.h>
 #if ((' ' & 0x0FF) == 0x020)
 # define ISLOWER(c) ('a' <= (c) && (c) <= 'z')
 # define TOUPPER(c) (ISLOWER(c) ? 'A' + ((c) - 'a') : (c))
--- index_main.c.orig	2000-06-13 03:44:11.000000000 -0500
+++ index_main.c	2021-02-25 16:38:49.000000000 -0600
@@ -1,3 +1,5 @@
+#include <string.h>
+
 #include "sgrep.h"
 
 /*
--- sgrep.h.orig	2001-03-15 06:14:48.000000000 -0600
+++ sgrep.h	2021-02-25 16:42:24.000000000 -0600
@@ -612,6 +612,7 @@
     FileList *file_list;
     const char *file_name;
 } IndexOptions;
+int index_query(IndexOptions *options, int argc, char *argv[]);
 void set_default_index_options(SgrepData *sgrep,IndexOptions *o);
 int create_index(const IndexOptions *options);
 int add_region_to_index(struct IndexWriterStruct *writer,
